duckduckgo/Android
# Improved bookmarks screen performace.

Task/Issue URL: https://app.asana.com/0/414730916066338/1202088634690441/f

### Description

Changes:

1/ Updated to insert all bookmarks in one go and all favorites in one go. 
2/ Removed the ListAdapter along with the DiffCallback from BookmarksAdapter / FavoritesAdapter. This was incredibly slow, taking more than 20 seconds to display a list, unlike a regular implementation of RecyclerView.Adapter.
3/ Removed the usage of findViewById() when updating the viewholder in both BookmarksAdapter and FavoritesAdapter. Updated the TwoLineListItem custom view to give access to the image.
4/ Updated the loadFavicon in both BookmarksAdapter and FavoritesAdapter to do all operations needed to find the favicon using the IO dispatcher. Updated the actual Glide call which loads the image to use the Main dispatcher and also forced cancelling image requests that are not needed (i.e the view is no longer on screen).
5/ Used StrictMode before and after the changes to prove the main thread is now free of unnecessary operations.

### Steps to test this PR

To see the issue:
- fresh install from develop
- import a large number of bookmarks / favorites. Example files are available in the asana task.
- while importing try to scroll up / down
- notice the app crashes

To test the fix:
- fresh install from this branch
- import a large number of bookmarks / favorites. Example files are available in the asana task.
- while importing try to scroll up / down
- the app should not crash
- additional tests: do a search in bookmarks

### NO UI changes
